app/waf/src/controllers/schema.ts

```diff
@@ -220,69 +106,180 @@ function buildRequestSchema(
   };
 }
 
-function buildResponseSchema(properties: object, example: object): object {
+function buildResponseSchema(
+  name: string,
+  properties: object,
+  example: object,
+): object {
+  let resp: {[key: string]: object} = {};
+  resp[name] = {
+    type: 'object',
+    properties: properties,
+    example: example,
+  };
+
   return {
     schema: {
       type: 'object',
-      properties: properties,
-      example: example,
+      properties: resp,
     },
   };
 }
 
 function buildCollectionResponseSchema(
+  name: string,
```

Comment:
Just feel that the usage of this function is little. return that object is more direct and clear.

Reply:
Why little? Every collection response api need this schema.

Reply:
1. The function is called only once.
2. What does the function do?
Only returns a wrapped object with `schema`. How about to re-define collectionResponse function with: 

```
  static collectionResponse(
    entity: typeof Entity,
    name: string,
    desc: string,
  ): object {
    let props = buildProperties(propertyMetadata(entity));
    return {
      description: desc,
      content: {
        'application/json': {
          schema: {
            type: 'object',
            properties: {
              [name]: {
                type: 'array',
                items: {
                  type: 'object',
                  properties: props.response,
                  example: props.responseExample,
                },
              }
            },
          },
        },
      },
    };
  }
```

That's is clear for us to see the whole body of the schema definition.

Reply:
I just think there's no need to define a separate function for constructing the schema, which it makes the schema not so human readable.

Reply:
That should be optional. It seem not very worth to doing this right way.

Reply:
Make it clear.